src/engine/DebugOverlay.EntityLabel.cs

```diff
@@ -0,0 +1,158 @@
+    {
+        labelsLayer = GetNode<Control>(EntityLabelsPath);
+        smallerFont = GD.Load<Font>("res://src/gui_common/fonts/Lato-Regular-Tiny.tres");
+    }
+
+    private void EntityLabelEnterTree()
+    {
+        var rootTree = GetTree();
+        rootTree.Connect("node_added", this, nameof(OnNodeAdded));
+        rootTree.Connect("node_removed", this, nameof(OnNodeRemoved));
+    }
+
+    private void EntityLabelExitTree()
+    {
+        var rootTree = GetTree();
+        rootTree.Disconnect("node_added", this, nameof(OnNodeAdded));
+        rootTree.Disconnect("node_removed", this, nameof(OnNodeRemoved));
+    }
+
+    private void EntityLabelProcess()
+    {
+        if (activeCamera is not { Current: true })
+            activeCamera = GetViewport().GetCamera();
+
+        if (showEntityLabels)
+        {
+            UpdateEntityLabels();
+        }
+    }
+
+    private void UpdateEntityLabels()
+    {
+        if (activeCamera == null)
+            return;
+
+        foreach (var pair in entityLabels)
+        {
+            var body = pair.Key;
+            var label = pair.Value;
+
+            // Update names
+            if (label.Text.Empty())
+            {
+                switch (body)
+                {
+                    case Microbe microbe:
+                    {
+                        if (microbe.Species != null!)
+                        {
+                            label.Text =
+                                $"[{microbe.Name}:{microbe.Species.Genus[0]}.{microbe.Species.Epithet.Left(4)}]";
+                        }
+
+                        break;
+                    }
+
+                    case FloatingChunk chunk:
+                    {
+                        label.Text = $"[{chunk.Name}:{chunk.ChunkName}]";
+                        break;
+                    }
+
+                    default:
+                    {
+                        label.Text = $"[{body.Name}]";
+                        break;
+                    }
+                }
+            }
+
+            label.RectPosition = activeCamera.UnprojectPosition(body.Transform.origin);
+        }
+    }
+
+    private void UpdateLabelOnMicrobeDeath(Microbe microbe)
+    {
+        if (entityLabels.TryGetValue(microbe, out var label))
+            label.Set("custom_colors/font_color", new Color(1.0f, 0.3f, 0.3f));
+    }
+
+    private void OnNodeAdded(Node node)
+    {
+        if (node is not RigidBody body)
+            return;
+
+        var label = new Label();
+        labelsLayer.AddChild(label);
+        entityLabels.Add(body, label);
+
+        switch (body)
+        {
+            case Microbe microbe:
+            {
+                microbe.OnDeath += UpdateLabelOnMicrobeDeath;
+                break;
+            }
+
+            case FloatingChunk:
+            case AgentProjectile:
+            {
+                label.AddFontOverride("font", smallerFont);
+                break;
+            }
```

Comment:
Should simply use `default` keyword as it doesn't matter whether this is FloatingChunk or AgentProjectile. Or you could even use this font for all entity types, not sure why Microbe is excluded.

Reply:
It's just because floating chunks always overlap each other, making it very difficult to read. Microbes are separated.